### DISCUSSION TITLE
Use GPU instead of integrated graphics

### DISCUSSION THREAD

**Discussion by ghost** [Schneegans/Burn-My-Windows#103] [Q&A] [answered]
Hello @Schneegans !

I'm having a hard time with the effects, specifically with the fire effect, when the windows are full (or somewhat full) screen, it uses the integrated graphics (Mesa Intel HD Graphics 630 (KBL GT2)) to run the animations, and boy, they **are** slow, but I have a GTX 1050 Mobile available.

I'm a layman in things about compositor, animations, etc. Is it possible to offload these animations to the GPU?

Maybe I missed something in [NVIDIA - Arch Linux Wiki](https://wiki.archlinux.org/title/NVIDIA), I'm not sure.

Thank you for this extension, I'll buy you another beer!

**Comment by ghost** [comment]
Maybe this will help me: https://wiki.archlinux.org/title/PRIME#PRIME_render_offload

I don't think that sending animations to dedicated GPU (in my case NVIDIA [Prime]) is the developer's "job". I think that are some settings missing in my setup.

**Comment by ghost** [comment]
On another note (and it's just a thought): do you think that even if it's using the integrated graphics, it should be slow or require more processing? Are there any optimizations that can be done to mitigate this?

It happens with the default and tweaks of settings

Just out of curiosity, I'm not demanding, after all, I'm not experienced in this.

**Comment by Schneegans** [comment, marked as answer]
Hi there! First of all: Thanks for the donation, beer is always appreciated :wink: 

It is definitely not possible to use the NVIDIA card for _only_ the effects. However, it should be possible to run GNOME Shell entirely on it. If you have the proprietary drivers installed, you have to make sure to select the "Performance Mode". Like this:

![Screenshot from 2022-02-14 20-21-37](https://user-images.githubusercontent.com/829942/153931757-a9642888-c11d-4088-a2f2-15680197f70c.png)

For me (currently on Ubuntu 21.10), this does not work on Wayland yet (actually it should be working with the most recent drivers, AFAIK). Also, for me GNOME Shell on Wayland + integrated graphics actually runs with less micro stutters than on X11 + NVIDIA...

Performance could be improved, I guess. The main problem is the computation of the noise functions in the shader. This is rather computational expensive. This could be improved by using precomputed noise textures, however this won't work on GNOME 3.36 and 3.38.
